pkg/stub/handler.go

```diff
@@ -247,3 +255,179 @@ func (h *Handler) ensureRouterDeleted(ci *ingressv1alpha1.ClusterIngress) error
 	}
 	return nil
 }
+
+// syncOperatorStatus computes the operator's current status and therefrom
+// creates or updates the ClusterOperator resource for the operator.
+func (h *Handler) syncOperatorStatus() {
```

Comment:
Maybe move all this stuff and the new stuff in util into a new file (`status.go`)?